physics/physics.js

```diff
@@ -0,0 +1,505 @@
+      var ny = normalizeBinormalComponent(corner1[1] - corner2[1]);
+
+      if (collideWithLine(corner0[0], corner0[1], corner1[0], corner1[1], nx, ny, foil.vx, foil.vy, sphere))
+        return;
+    }
+
+    var corner = Object();
+    corner.radius = 0.1;
+    corner.mass = 50000; // really big.
+
+    for (var i = 0; i < corners.length; ++i) {
+      corner.dx = corners[i][0];
+      corner.dy = corners[i][1];
+      corner.vx = foil.vx;
+      corner.vy = foil.vy;
+      if (collideWithSphere(sphere, corner))
+        return;
+    }
+  }
+
+  function updateFoilProperties(foil) {
+    var dy = getAnimatedPosition(foil) + foil.top;
+    foil.vy = dy - foil.dy;
+    foil.dy = dy;
+  }
+
+  function processCollisions() {
+    for (var i = 0; i < foils.length; ++i)
+      updateFoilProperties(foils[i]);
+    for (var i = 0; i < spheres.length; ++i) {
+      for (var j = 0; j < foils.length; ++j)
+        collideWithFoil(spheres[i], foils[j]);
+      for (var j = i + 1; j < spheres.length; ++j)
+        collideWithSphere(spheres[i], spheres[j]);
+    }
+  }
+
+  function resetSpheres() {
+    for (var i = 0; i < spheres.length; ++i) {
+      var sphere = spheres[i];
+      var padding = 400;
+      if (sphere.dy > viewportHeight + padding) {
+        sphere.dx = sphere.initialX;
+        sphere.dy = sphere.initialY - 0.5 * viewportHeight;
+        sphere.vy = 0;
+        sphere.vx *= epsilon;
+        if (Math.abs(sphere.vx) > 100)
+          sphere.vx = 0;
+      }
+    }
+  }
+
+  function updateLogo() {
+    logoAngle += logoSpeed;
+    while (logoAngle > 360)
+      logoAngle -= 360;
+
+    logo.style.webkitTransform = "translateZ(0) rotate(" + logoAngle + "deg)";
+  }
+
+  scope.tick = function(timestamp) {
+    var dt = 1.0;
+    console.log("tick: " + timestamp);
+    if (scope.lastTimestamp) {
+      dt = (timestamp - scope.lastTimestamp) * 60;
+    }
+    scope.lastTimestamp = timestamp;
+
+    if (isMain()) {
+      updateLogo();
+    } else {
+      updatePositions(dt);
+      processCollisions();
+      updatePositions(dt);
+      applyGravity(dt);
+      resetSpheres();
+    }
+    scope.requestAnimationFrame(tick);
+  }
+
+  function createFoil(dx, dy, width, height, delay) {
+    var foilActor = document.createElement("div");
+    foilActor.className = "foil";
+    foilActor.style.left = dx + "px";
+    foilActor.style.top = dy + "px";
+    foilActor.style.width = width;
+    foilActor.style.height = height;
+    foilActor.style.webkitAnimation = "foil-motion 1s " + delay + "ms infinite alternate";
+    document.body.appendChild(foilActor);
+
+    var foil = new Object();
+    foil.actor = foilActor;
+    foil.left = dx;
+    foil.top = dy;
+    foil.width = width;
+    foil.height = height;
+    foil.dx = foil.left;
+    foil.dy = foil.top;
+    foil.vx = 0;
+    foil.vy = 0;
+
+    return foil;
+  }
+
+  function busySleep(duration) {
+    var start = new Date();
+    while ((new Date()).getTime() - start.getTime() < duration);
```

Comment:
As far as I know, we are landing *a subset* of the Houdini specs behind a flag in Canary soon-ish :tm: Don’t know exact dates myself, sorry.